sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -132,6 +132,30 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
     this
   }
 
+  def parquet(path: String): StreamingQuery = {
+    format("parquet")
+      .option("path", path)
+      .start()
+  }
+
+  def json(path: String): StreamingQuery = {
+    format("json")
+      .option("path", path)
+      .start()
+  }
+
+  def csv(path: String): StreamingQuery = {
+    format("csv")
+      .option("path", path)
+      .start()
+  }
+
+  def text(path: String): StreamingQuery = {
+    format("text")
+      .option("path", path)
+      .start()
+  }
```

Comment:
Unfortunately, we are recommending `format` over this `parquet/json/csv/text`. IIRC, this is not added due to that reason. For example, this doesn't cover `orc` and external `avro`, too.

cc @gatorsmile , @cloud-fan , @srowen , @HyukjinKwon .

Reply:
I just thought it would be good to have access to the same methods. Also I was thinking that `.start` could be out of the `.parquet/json/csv/text` to keep making people use `start` to start streaming queries.
About `orc` and `avro` you are right, they are missing right now, but they could be added.

Reply:
Yes. You can easily create an implicit class with these extensions if you need them.

Reply:
Should I close it then? @cloud-fan @dongjoon-hyun  @HyukjinKwon

Reply:
yes, please.

Reply:
Yesh

Reply:
Yeah, thanks for closing.